### PR TITLE
user_service_Event_3

### PR DESCRIPTION
**Pull request by KateAle**

### CODE REVIEW

File: src/test/java/school/faang/user_service/service/UserServiceTest.java
Comment (pablo023) [inline src/test/java/school/faang/user_service/service/UserServiceTest.java:1]: @ExtendWith(MockitoExtension.class)
    @DisplayName("UserService CSV Processing Tests")
    public class UserServiceTest {

    @Mock
    private UserRepository userRepository;

    @Mock
    private CountryRepository countryRepository;

    @Mock
    private CsvMapper csvMapper;

    @Mock
    private PersonToUserMapper userMapper;

    @InjectMocks
    private UserService userService;

    private InputStream createMockInputStream() {
        String csvData = "firstName,lastName,email,phone,country\n" + "John,J,john.j@example.com,1234567890,USA";
        return new ByteArrayInputStream(csvData.getBytes(StandardCharsets.UTF_8));
    }

    @Test
    @DisplayName("Should successfully process a valid CSV")
    public void testProcessCsv_ValidInput() throws Exception {
        setupMockBehaviorForValidInput();
        InputStream inputStream = createMockInputStream();
        userService.processCsv(inputStream);

        verify(csvMapper, times(1)).readerFor(Person.class);
    }

    @Test
    @DisplayName("Should fail while parsing CSV and never save anything to repository")
    public void testProcessCsv_ExceptionHandling() {
        InputStream inputStream = createMockInputStream();
        userService.processCsv(inputStream);
        verify(userRepository, never()).save(any(User.class));
    }

    private void setupMockBehaviorForValidInput() throws Exception {
        CsvSchema schema = CsvSchema.emptySchema().withHeader();
        when(csvMapper.schemaFor(Person.class)).thenReturn(schema);

        Address address = new Address();
        address.setCountry("USA");

        ContactInfo contactInfo = new ContactInfo();
        contactInfo.setEmail("john.j@example.com");
        contactInfo.setPhone("1234567890");
        contactInfo.setAddress(address);

        Person mockPerson = new Person();
        mockPerson.setFirstName("John");
        mockPerson.setLastName("J");
        mockPerson.setContactInfo(contactInfo);

        Country country = new Country();
        country.setId(1L);
        country.setTitle("USA");

        User user = new User();
        user.setUsername(userMapper.combineName(mockPerson.getFirstName(), mockPerson.getLastName()));
        user.setEmail(contactInfo.getEmail());
        user.setPhone(contactInfo.getPhone());
        user.setCountry(country);
        user.setPassword("password");

        var objectReader = Mockito.mock(com.fasterxml.jackson.databind.ObjectReader.class);
        when(csvMapper.readerFor(Person.class)).thenReturn(objectReader);
    }
}

**Review by pablo023 — CHANGES_REQUESTED** [review]

**Comment by pablo023** [comment]
Вот тебе работающие тесты. Пофикси тесты и конфликты, и давай за новую задачу)

File: src/test/java/school/faang/user_service/service/UserServiceTest.java
Comment (KateAle) [reply src/test/java/school/faang/user_service/service/UserServiceTest.java:1]: работает)) спасибо!)

Comment (pablo023) [reply src/test/java/school/faang/user_service/service/UserServiceTest.java:1]: Работает) Конфликты только пофикси теперь)

Comment (KateAle) [reply src/test/java/school/faang/user_service/service/UserServiceTest.java:1]: я так и не поняла, где конфликты возникли, сервер взлетел, билт тоже...ветку я подтянула

**Pull request closed without merging**